fabric-server-languages-v1/src/main/java/net/fabricmc/fabric/mixin/server/languages/v1/MixinLanguage.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.server.languages.v1;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.JsonParseException;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.util.Language;
+
+import net.fabricmc.loader.api.FabricLoader;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.function.BiConsumer;
+import java.util.HashMap;
+
+@Mixin(Language.class)
+public class MixinLanguage {
+	@Shadow
+	@Final
+	private static Logger LOGGER;
+	@Shadow
+	@Final
+	private static String DEFAULT_LANGUAGE;
+
+	@Shadow
+	private static void load(InputStream inputStream, BiConsumer<String, String> entryConsumer) {
+	}
+
+	private static void loadModLangFile(String modId, BiConsumer<String, String> biConsumer) {
+		String path = "/assets/" + modId + "/lang/" + DEFAULT_LANGUAGE + ".json";
+
+		try (InputStream inputStream = Language.class.getResourceAsStream(path)) {
+			if (inputStream == null) return;
+			load(inputStream, biConsumer);
```

Comment:
For the load process, I would recommend something like this:
(Use [`ModContainer.getPath()`](https://github.com/FabricMC/fabric-loader/blob/5692a3013ed1b2a7451bcfaac5a0ac93f8d1b10e/src/main/java/net/fabricmc/loader/api/ModContainer.java#L60))
```java
Path path = modContainer.getPath("assets/" + modContainer.getMetadata().getId() + "/lang/" + DEFAULT_LANGUAGE + ".json");

if (!Files.exists(path)) {
	return;
}

try (InputStream inputStream = Files.newInputStream(path)) {
	load(inputStream, biConsumer);
```

Another thing to note is encoding; we should support utf-8 for mod json files, but we don't know what encoding minecraft expects in this method.

Reply:
I pushed a commit that uses nio and ModContainer.getPath(). I also looked at the vanilla method `load(... inputStream, ... biConsumer)` and it does use UTF-8.

> I would get ModResourcePacks instead, it would avoid duplication of logic and allow to be more true to how language loading actually works in MC (which is load from every resource packs, from every namespace, the lang file).

Could you help guide me to what this implementation would look like? I'm not sure what's being suggested here.